tests/pytype_test.py

```diff
@@ -65,27 +87,49 @@ def communicate(self):
 
 def pytype_test(args):
     try:
-        PytdRun(["-h"]).communicate()
+        BinaryRun(['pytd', '-h']).communicate()
     except OSError:
-        print("Cannot run pytd. Did you install pytype?")
+        print('Cannot run pytd. Did you install pytype?')
         return 0, 0
 
-    wanted = re.compile(r"stdlib/.*\.pyi$")
-    skipped = re.compile("(%s)$" % "|".join(load_blacklist()))
-    files = []
+    blacklist, parse_only = load_blacklist()
+    wanted = re.compile(r'stdlib/.*\.pyi$')
+    skipped = re.compile('(%s)$' % '|'.join(blacklist))
+    parse_only = re.compile('(%s)$' % '|'.join(parse_only))
+
+    pytype_run = []
+    pytd_run = []
 
-    for root, _, filenames in os.walk("stdlib"):
+    for root, _, filenames in os.walk('stdlib'):
         for f in sorted(filenames):
             f = os.path.join(root, f)
-            if wanted.search(f) and not skipped.search(f):
-                files.append(f)
+            if wanted.search(f):
+                if parse_only.search(f):
+                    pytd_run.append(f)
+                elif not skipped.search(f):
+                    pytype_run.append(f)
 
     running_tests = collections.deque()
     max_code, runs, errors = 0, 0, 0
-    print("Running pytype tests...")
+    files = pytype_run + pytd_run
     while 1:
         while files and len(running_tests) < args.num_parallel:
-            test_run = PytdRun([files.pop()], dry_run=args.dry_run)
+            f = files.pop()
+            if f in pytype_run:
+                temp = tempfile.NamedTemporaryFile(delete=False)
+                temp.close()  # Windows compat.
+                test_run = BinaryRun(
+                    ['pytype',
+                     '--typeshed-location=%s' % os.getcwd(),
+                     '--module-name=%s' % _get_module_name(f),
+                     '--convert-to-pickle=%s' % temp.name,
```

Comment:
Do you need the temporary file? Or could you just use `os.devnull`?

Reply:
Done